### PR TITLE
networkmanager: 1.20.8 -> 1.22.4, modemmanager: 1.10.6 -> 1.12.2

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.22.0/NEWS
https://gitlab.freedesktop.org/mobile-broadband/ModemManager/blob/1.12.2/NEWS

Experimental features in NetworkManager are not enabled in this PR.

###### Things done
Built on master.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
> * DHCP: switch "internal" DHCPv4 plugin from code based on systemd to use nettools' n-dhcp4 library.

Do we support that? Are nettools not needed?

I found https://gitlab.freedesktop.org/NetworkManager/NetworkManager/commit/d3c7083f976f0e769d7f0b55c3024084e46667b1 but it does not make it very clear.

**Comment by worldofpeace** [comment]
@jtojnar I think this is `internal` which is code in NetworkManager.
`dhcp` defaulted to `internal` since the update to `1.20`. However I'm already seeing bugreports and fixes because of using this code https://gitlab.freedesktop.org/NetworkManager/NetworkManager/merge_requests/368/diffs.

**Comment by jtojnar** [comment]
Looks like the library is vendored so I retract my comment: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/tree/master/shared/n-dhcp4

**Comment by worldofpeace** [comment]
I full confidence the 1.22.2 is probably more fab.

**Comment by worldofpeace** [comment]
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/1.22.4/NEWS, ooh :rofl: 
There's a fake hash in here, I should probably fix that in the next push.

**Comment by worldofpeace** [comment]
Built on d7627dd2df9e40ace0d494f25d05f7bcbb18151c
```
/nix/store/qzr93yjmhnglkvbqfh6184bpiybpvics-network-manager-1.22.4
/nix/store/svcngl3lvnswjrb2cm01bbs1a6s1hyhw-modem-manager-1.12.2
```

File: pkgs/development/python-modules/python-dbusmock/default.nix [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
   nose, dbus, dbus-python, pygobject3,
   which, pyflakes, pycodestyle, bluez, networkmanager
 }:
 
 buildPythonPackage rec {
   pname = "python-dbusmock";
-  version = "0.18.3";
+  version = "0.19";
 
-  src = fetchPypi {
```
Comment (jtojnar) [inline pkgs/development/python-modules/python-dbusmock/default.nix]: Could you open issue against upstream that the release is missing from PyPI?

Comment (jonringer) [reply pkgs/development/python-modules/python-dbusmock/default.nix]: https://github.com/martinpitt/python-dbusmock/issues/58

**Comment by worldofpeace** [comment]
Ok, I've built everything and the only issue was with python-dbusmock which is fixed with the update.

File: pkgs/development/python-modules/python-dbusmock/default.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/development/python-modules/python-dbusmock/default.nix]: Thanks @jonringer

**Pull request merged**